core/model/auth.go

```diff
@@ -202,6 +202,38 @@ func (ls LoginSession) CurrentRefreshToken() string {
 	return ls.RefreshTokens[numTokens-1]
 }
 
+// PreviousRefreshToken returns the previous refresh token (second to last element of RefreshTokens)
+func (ls LoginSession) PreviousRefreshToken() string {
+	numTokens := len(ls.RefreshTokens)
+	if numTokens <= 1 {
+		return ""
```

Comment:
I think we should be careful here about returning an empty string as a default value. It may be that we are validating that the refresh token sent by the user is not an empty string elsewhere in the code, but I think it may be worth either changing this to return nil in this case, or adding an explicit check that the refresh token string is not empty wherever we use it.

I see that our `CurrentRefreshToken` function above also had a similar issue so I think it's unlikely that this is currently an issue but just thought I'd flag it since I noticed it.